### PR TITLE
Fix broken coming soon template in dataview

### LINKED ISSUE #49699: Broken coming soon template in dataview
### Issue Description

Originally reported in p1719905292036319-slack-C01SFMVEYAK and p1721309784520059-slack-C06DJHTE04U

The coming soon entire site thumbnail in templates dataview is broken.

image

The cause might have been because we set `vh` to `width` and `height` of the template component.

### Acceptance Criteria

We're not sure what's the most appropriate fix, but Gutenberg upstream fix will be the most ideal. Otherwise, there's a couple potential workarounds:

1. Dynamically change to `%` width & height if it’s rendered from a dataview
2. Amend the design so it doesn't depend on having to use `vh`, but the template needs to stretch to fill the screen both horizontally and vertically

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/49699

This PR fixes the broken "coming soon" template in the dataview. The issue is that the preview iframe [height is set dynamically based on the content height](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-preview/auto.js#L101), while our pattern height is set to 100vh, causing dependency issues. To fix this, we could set the content height to a fixed value, but this isn't ideal because the content height varies. Instead, the best solution is to set the aspect ratio of the content body to 1/1, making the height and width equal, since the preview block is a square. This allows the preview iframe to calculate the height correctly. However, if Gutenberg changes the preview block to a rectangle in the future, this solution may need updating. But for now, it's the best approach IMO.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site using this branch.
2. Skip Core Profiler
3. Go to `/wp-admin/site-editor.php?postType=wp_template`
4. Search for the `Coming Soon` template
5. Observe the preview should be displayed correctly
6. Click on the template and do some customizations
7. Observe the preview should be still displayed correctly

Screenshot 2024-07-22 at 15 21 09
Screenshot 2024-07-22 at 15 21 44

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/fix-broken-coming-soon-template [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Set coming soon preview body aspect ratio to 1/1 to fix borken template
```
Comment (ilyasfoo) [inline plugins/woocommerce/changelog/fix-broken-coming-soon-template]:
```suggestion
Set coming soon preview body aspect ratio to 1/1 to fix broken template
```

**Review by ilyasfoo — APPROVED** [review]
Thanks for the fix @chihsuan. There's an issue if the site already has customization before installing this update. To reproduce:

1. Set up a JN site using `trunk` branch
2. Go to `Appearance > Editor > Templates > Page: Coming Soon`
3. Edit the text in the middle and save
4. Go to `WooCommerce > Live Branches` and Install & activate this branch
5. Go to `Appearance > Editor > Templates > Page: Coming Soon`
6. Look at coming soon template and observe the "Attempt recovery" text and problematic render


https://github.com/user-attachments/assets/538a88e4-80ea-45b1-812a-5faabc08c9ad

I'm wondering if we already have a solution for updating blocks. If it's possible to apply it with ours, otherwise, this fix is looking good anyway.

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀 

TIL 😄  -- aspect-ratio

**Comment by chihsuan** [comment]
> I'm wondering if we already have a solution for updating blocks. If it's possible to apply it with ours, otherwise, this fix is looking good anyway.
 
Hey @ilyasfoo, I did some investigation, and it seems the standard way to update blocks is to deprecate the old block and create a new one or provide a “deprecated” version. I think we can do that for this block. However, I think it's better to move the styles out of the block so that we don't need to deprecate it every time we update the styles. I'll work on this in another PR today. Does that sound good to you?

> [**I’ve changed my block’s save behavior and old content now includes invalid blocks. How can I fix this?**](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#validation-faq)

> Refer to the guide on [Deprecated Blocks](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/) to learn more about how to accommodate legacy content in intentional markup changes.

**Comment by ilyasfoo** [comment]
> However, I think it's better to move the styles out of the block so that we don't need to deprecate it every time we update the styles. I'll work on this in another PR today. Does that sound good to you?

@chihsuan Yes, that sounds great to me. Good thinking to move out the styles 👍 

> I did some investigation, and it seems the standard way to update blocks is to deprecate the old block and create a new one or provide a “deprecated” version. I think we can do that for this block

I agree, thanks for investigating!

**Pull request merged**